webpack/assets/javascripts/react_app/components/bookmarks/form/index.js

```diff
@@ -1,52 +1,56 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { reduxForm } from 'redux-form';
 import { connect } from 'react-redux';
-import { required, length } from 'redux-form-validators';
+import { Formik } from 'formik';
 
 import { noop } from '../../../common/helpers';
 import Form from '../../common/forms/Form';
 import TextField from '../../common/forms/TextField';
 import * as FormActions from '../../../redux/actions/common/forms';
 import { translate as __ } from '../../../../react_app/common/I18n';
+import { required, maxLength, errorsToSentence } from '../../common/forms/validators';
 
-const submit = ({ name, query, publik }, dispatch, props) => {
-  const { submitForm, url, controller } = props;
-  const values = {
-    name,
-    query,
-    public: publik || false,
-    controller,
-  };
-
-  return submitForm({ url, values, item: 'Bookmark' });
-};
-
-const BookmarkForm = ({ handleSubmit, submitting, error, onCancel }) => (
-  <Form
-    onSubmit={handleSubmit(submit)}
-    onCancel={onCancel}
-    disabled={submitting}
-    submitting={submitting}
-    error={error}
+const BookmarkForm = ({ url, submitForm, controller, onCancel, initialValues }) => (
+  <Formik
+    onSubmit={(values, actions) => {
+      submitForm({ url, values: { ...values, controller }, item: 'Bookmark' }).catch(exception => {
+        actions.setSubmitting(false);
+        actions.setErrors(Object.keys(exception.errors).reduce((memo, key) => {
+          const errorMessages = exception.errors[key]
+          memo[key] = errorMessages ? errorMessages.join(', ') : errorMessages;
+          return memo;
+        }, {}));
+      });
+    }}
+    initialValues={initialValues}
   >
-    <TextField
-      name="name"
-      type="text"
-      required="true"
-      label={__('Name')}
-      validate={[required(), length({ max: 254 })]}
-    />
-    <TextField
-      name="query"
-      type="textarea"
-      required="true"
-      label={__('Query')}
-      inputClassName="col-md-8"
-      validate={[required(), length({ max: 4096 })]}
-    />
-    <TextField name="publik" type="checkbox" label={__('Public')} />
-  </Form>
+    {({ handleSubmit, isSubmitting, isValid }) => (
+        <Form
+          onSubmit={handleSubmit}
+          onCancel={onCancel}
+          disabled={isSubmitting}
+          submitting={isSubmitting}
+        >
+          <TextField
```

Comment:
formik has its own `Form` and `ErrorMessage` components which reducing boilerplate, does it make sense to consume these instead of our own implementation ?

Reply:
I use `Form` from formik now, but it does not provide significant benefits. `ErrorMessage` provides just a basic wiring, we already have a highly customized way to display errors.